### PR TITLE
FAI-11608: GitHub Tags Source + Stream Converter

### PR DESCRIPTION
**Pull request by matiaslcoulougian**
## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: faros-airbyte-common/src/github/types.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import {Octokit} from '@octokit/rest';
 import {GetResponseDataTypeFromEndpointMethod} from '@octokit/types';
+import {exec} from 'child_process';
```
Comment (dbruno21) [inline faros-airbyte-common/src/github/types.ts:3]:
```suggestion
```

File: sources/github-source/src/streams/faros_tags.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    yield* github.getTags(org, repo, cutoffDate);
+  }
+
+  getUpdatedState(
+    currentStreamState: StreamState,
+    latestRecord: Tag,
+    slice: RepoStreamSlice
+  ): StreamState {
+    const latestRecordCutoff = Utils.toDate(
+      latestRecord?.commit.committedDate ?? 0
```
Comment (dbruno21) [inline sources/github-source/src/streams/faros_tags.ts:47]: Can commit be null / undefined?

File: sources/github-source/src/github.ts [hunk 3]
```diff
@@ -899,6 +903,40 @@ export abstract class GitHub {
...
+    org: string,
+    repo: string,
+    cutoffDate?: Date
+  ): AsyncGenerator<Tag> {
+    const iter = this.octokit(org).graphql.paginate.iterator<RepoTagsQuery>(
+      REPOSITORY_TAGS_QUERY,
+      {
+        owner: org,
+        repo,
+        per_page: this.pageSize,
```
Comment (dbruno21) [inline sources/github-source/src/github.ts:916]: There's no param to filter by date, right?

File: sources/github-source/src/github.ts [hunk 4]
```diff
@@ -899,6 +903,40 @@ export abstract class GitHub {
...
+      for (const tag of res.repository.refs.nodes) {
+        let commit: TagsQueryCommitNode;
+        if (tag.target.type === 'Commit') {
+          commit = tag.target;
+        }
+        if (tag.target.type === 'Tag' && tag.target.target.type === 'Commit') {
+          commit = tag.target.target;
+        }
+        if (cutoffDate && Utils.toDate(commit.committedDate) <= cutoffDate) {
+          continue;
```
Comment (dbruno21) [inline sources/github-source/src/github.ts:929]: There's no way / param to sort by date, right? If there is or results are already sorted this should be a break;

Comment (dbruno21) [reply sources/github-source/src/github.ts:929]: How does feeds handle the incremental mode?

File: sources/github-source/src/streams/faros_tags.ts (hunk 2 above)
Comment (matiaslcoulougian) [reply sources/github-source/src/streams/faros_tags.ts:47]: Here it should not be null. But target could be null, will handle that case [here](https://github.com/faros-ai/airbyte-connectors/pull/1615/files#diff-c3ec475eeb0ddb699ccf89b07eb40f24741bc8974d505f77035e1a8d27f4e563R922):

File: sources/github-source/src/github.ts (hunk 3 above)
Comment (matiaslcoulougian) [reply sources/github-source/src/github.ts:916]: Nope

File: sources/github-source/src/github.ts (hunk 4 above)
Comment (matiaslcoulougian) [reply sources/github-source/src/github.ts:929]: Fees do not sync this incrementally, all tags are always pulled and written

Comment (matiaslcoulougian) [reply sources/github-source/src/github.ts:929]: Removed incremental sync so we don't miss any tags added to commits committed before cutoffDate.

**Review by dbruno21 — APPROVED** [review]

**Pull request merged**